### DISCUSSION TITLE
Pre-work for migrating to a fullstack framework

### DISCUSSION THREAD

**Discussion by mariojsnunes** [ONEARMY/community-platform#3141] [Feature request / Ideas]
We should discuss the steps to migrate to a fullstack framework which is being decided here https://github.com/ONEARMY/community-platform/discussions/1646.

Since the pages will be server-side rendered, most scenarios won't use a frontend store/cache.
This is important because most of the current code uses stores.

What I think needs to happen:
- Ensure firestore has appropriate indexes to allow efficient queries through filtering and sorting
- Migrate stores to "services" - methods to query the backend (which would also be used with ssr)

This is a big change, but can be done gradually.

I could start working on this (for the questions module?), but would like to hear your thoughts first.

**Comment by thisislawatts** [comment]
Thanks for the suggestion @mariojsnunes, this sounds reasonable to me but I think we are still lacking some internal alignment on how/when to prioritise this work. Which is why there has been relatively little movement on https://github.com/ONEARMY/community-platform/discussions/1646

I have done some experimentation targeting Cloud Run as a target for offloading requests:
https://github.com/ONEARMY/community-platform/compare/spike/reader-app

Specifically here is a sketch with Astro's API endpoints. 
https://github.com/ONEARMY/community-platform/commit/7bc61c4c54b43a048ab44d528d4ed53c3ba11cdc#diff-3160edfb06e8a9c4f44aab3509c30d008d670694e8cb67b812cde8d891645835

**Comment by mariojsnunes** [comment]
The migration was completed in late 2024
